innopoints/views/activity.py

```diff
@@ -108,16 +108,17 @@ def patch(self, project_id, activity_id):
         if not updated_activity.draft and not updated_activity.is_complete:
             abort(400, {'message': 'Incomplete activities cannot be marked as non-draft.'})
 
-        for application in updated_activity.applications:
-            if application.status != ApplicationStatus.rejected:
-                application.actual_hours = updated_activity.working_hours
-
         if activity.fixed_reward and activity.working_hours != 1:
             abort(400, {'message': 'Cannot set working hours for fixed activities.'})
 
         if not activity.fixed_reward and activity.reward_rate != IPTS_PER_HOUR:
             abort(400, {'message': 'The reward rate for hourly activities may not be changed.'})
 
+        with db.session.no_autoflush:
```

Comment:
I don't get what is autoflush and why did it not cause problems elsewhere?

Reply:
Autoflush happens before queries. It's like an implicit `commit()`. In this case we make queries to applications, which causes it to commit changes which may violate integrity